Failed to link with multiple definition of X

# Problem

mruby-acme-client fails to link with multiple definition of X errors when -fno-common is enabled.

Since gcc version 10, -fno-common is enabled by default, so multiple definition of X error occurs.
It seems that it is well known issue [1] on Fedora 32, for example.

[1] https://src.fedoraproject.org/rpms/redhat-rpm-config/blob/master/f/buildflags.md#legacy-fcommon

It needs to eliminate multiple definition correctly.

# Expected

No linker error with -fno-common.

# How to reproduce

build on Fedora 32 or build by the following steps.

1. edit build_config.rb to enable `-fno-common`.
2. Run rake

```
diff --git a/build_config.rb b/build_config.rb
index 62a3dd2..4f7fb26 100644
--- a/build_config.rb
+++ b/build_config.rb
@@ -3,4 +3,7 @@ MRuby::Build.new do |conf|
   conf.gembox 'default'
   conf.gem File.expand_path(File.dirname(__FILE__))
   conf.enable_test
+  conf.cc do |cc|
+    cc.flags << '-fno-common'
+  end
 end
```

# Actual

Here is the partial linker error log.

```
AR    build/host/lib/libmruby.a 
LD    build/host/bin/mrbtest 
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/work/nginx/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `
mX509'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_x509.h:3: first defined here
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:6: multiple definition of
 `eDigestError'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:6: first define
d here
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:5: multiple definition of
 `mDigest'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:5: first defined her
e
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl_bn.o):/work/nginx/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX
509'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_x509.h:3: first defined here
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl_bn.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `
eDigestError'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:6: first defined 
here
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl_bn.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `
mDigest'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:5: first defined here
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl.o):/work/nginx/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509
'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_x509.h:3: first defined here
/bin/ld: /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(ossl.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDi
gestError'; /work/nginx/mruby-acme-client/mruby/build/host/lib/libmruby.a(acme_client.o):/work/nginx/mruby-acme-client/src/ossl_digest.h:6: first defined her
e
```

build log is also attached.
[error.log](https://github.com/pyama86/mruby-acme-client/files/4323527/error.log)

i can confirm the issue is also existing with fedora 33:

```
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509req.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509req.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509req.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509name.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509name.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509name.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509ext.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509ext.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509ext.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509crl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509crl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509crl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509cert.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509cert.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509cert.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509attr.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509attr.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509attr.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_digest.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_digest.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_digest.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_config.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_config.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_config.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bn.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bn.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bn.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bio.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bio.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bio.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey_rsa.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey_rsa.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey_rsa.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
collect2: Fehler: ld gab 1 als Ende-Status zurück
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_asn1.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509req.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509req.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509req.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509name.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509name.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509name.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509ext.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509ext.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509ext.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509crl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509crl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509crl.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509cert.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509cert.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509cert.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509attr.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509attr.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_x509attr.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_digest.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_digest.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_digest.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_config.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_config.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_config.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bn.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bn.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bn.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bio.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bio.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_bio.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey_rsa.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: multiple definition of `mX509'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_x509.h:3: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey_rsa.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: multiple definition of `eDigestError'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:6: first defined here
/usr/bin/ld: /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(ossl_pkey_rsa.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: multiple definition of `mDigest'; /tmp/ngx_mruby.BNx2/mruby/build/host/lib/libmruby.a(acme_client.o):/tmp/ngx_mruby.BNx2/mruby/build/repos/host/mruby-acme-client/src/ossl_digest.h:5: first defined here
collect2: Fehler: ld gab 1 als Ende-Status zurück
rake aborted!
Command failed with status (1): [gcc  -o "/tmp/ngx_mruby.BNx2/mruby/build/h...]
/tmp/ngx_mruby.BNx2/mruby/lib/mruby/build/command.rb:39:in `_run'
/tmp/ngx_mruby.BNx2/mruby/lib/mruby/build/command.rb:44:in `rescue in _run'
/tmp/ngx_mruby.BNx2/mruby/lib/mruby/build/command.rb:40:in `_run'
/tmp/ngx_mruby.BNx2/mruby/lib/mruby/build/command.rb:199:in `run'
/tmp/ngx_mruby.BNx2/mruby/Rakefile:80:in `block (4 levels) in <top (required)>'
/home/criess/.gem/ruby/gems/rake-13.0.0/exe/rake:27:in `<top (required)>'

Caused by:
Command failed with status (1): [gcc  -o "/tmp/ngx_mruby.BNx2/mruby/build/h...]
/tmp/ngx_mruby.BNx2/mruby/lib/mruby/build/command.rb:41:in `_run'
/tmp/ngx_mruby.BNx2/mruby/lib/mruby/build/command.rb:199:in `run'
/tmp/ngx_mruby.BNx2/mruby/Rakefile:80:in `block (4 levels) in <top (required)>'
/home/criess/.gem/ruby/gems/rake-13.0.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => default => all => /tmp/ngx_mruby.BNx2/mruby/bin/mruby-strip => /tmp/ngx_mruby.BNx2/mruby/build/host/bin/mruby-strip
(See full trace by running task with --trace)
make: *** [Makefile:78: build_mruby] Fehler 1
```

```
➭ lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	Fedora
Description:	Fedora release 33 (Thirty Three)
Release:	33
Codename:	ThirtyThree
```

--- a/src/ossl_digest.c
+++ b/src/ossl_digest.c
@@ -2,6 +2,9 @@
 //LICENSE: https://github.com/ruby/openssl/blob/master/LICENSE.txt
 #include "ossl.h"
 
+struct RClass *mDigest;
+struct RClass *eDigestError;
+
 #define SafeGetDigest(obj, ctx)                                                                    \
   do {                                                                                             \
     OSSL_Check_Kind((mrb), (obj), cDigest);                                                        \

--- a/src/ossl_digest.h
+++ b/src/ossl_digest.h
@@ -2,8 +2,6 @@
 //LICENSE: https://github.com/ruby/openssl/blob/master/LICENSE.txt
 #if !defined(_OSSL_DIGEST_H_)
 #define _OSSL_DIGEST_H_
-struct RClass *mDigest;
-struct RClass *eDigestError;
 
 void Init_ossl_digest(mrb_state *mrb);
 const EVP_MD *GetDigestPtr(mrb_state *mrb, mrb_value);

--- a/src/ossl_x509.c
+++ b/src/ossl_x509.c
@@ -2,6 +2,8 @@
 //LICENSE: https://github.com/ruby/openssl/blob/master/LICENSE.txt
 #include "ossl.h"
 
+struct RClass *mX509;
+
 void mrb_init_ossl_x509(mrb_state *mrb)
 {
   mX509 = mrb_define_module_under(mrb, mOSSL, "X509");

--- a/src/ossl_x509.h
+++ b/src/ossl_x509.h
@@ -1,6 +1,5 @@
 
 //LICENSE: https://github.com/ruby/openssl/blob/master/LICENSE.txt
-struct RClass *mX509;
 X509 *GetX509CertPtr(mrb_state *mrb, mrb_value VALUE);
 void Init_ossl_x509ext(mrb_state *mrb);
 void Init_ossl_x509cert(mrb_state *mrb);

--- a/src/ossl_x509attr.c
+++ b/src/ossl_x509attr.c
@@ -3,6 +3,7 @@
 #include "ossl.h"
 struct RClass *cX509Attr;
 struct RClass *eX509AttrError;
+extern struct RClass *mX509;
 
 static void ossl_x509attr_free(mrb_state *mrb, void *ptr)
 {

--- a/src/ossl_x509cert.c
+++ b/src/ossl_x509cert.c
@@ -3,6 +3,8 @@
 #include "ossl.h"
 struct RClass *cX509Cert;
 struct RClass *eX509CertError;
+extern struct RClass *mX509;
+
 #define GetX509(obj, x509)                                                                    \
   do {                                                                                             \
     x509 = DATA_PTR(obj);                                                                   \

--- a/src/ossl_x509crl.c
+++ b/src/ossl_x509crl.c
@@ -1,6 +1,7 @@
 
 //LICENSE: https://github.com/ruby/openssl/blob/master/LICENSE.txt
 #include "ossl.h"
+extern struct RClass *mX509;
 #define GetX509CRL(obj, crl)                                                                  \
   do {                                                                                             \
     crl = DATA_PTR(obj);                                                                     \

--- a/src/ossl_x509ext.c
+++ b/src/ossl_x509ext.c
@@ -4,6 +4,7 @@
 struct RClass *cX509Ext;
 struct RClass *cX509ExtFactory;
 struct RClass *eX509ExtError;
+extern struct RClass *mX509;
 
 #define MakeX509ExtFactory(obj, ctx)                                                               \
   do {                                                                                             \

--- a/src/ossl_x509name.c
+++ b/src/ossl_x509name.c
@@ -2,6 +2,7 @@
 //LICENSE: https://github.com/ruby/openssl/blob/master/LICENSE.txt
 #include "ossl.h"
 
+extern struct RClass *mX509;
 struct RClass *cX509Name;
 struct RClass *eX509NameError;
 

--- a/src/ossl_x509req.c
+++ b/src/ossl_x509req.c
@@ -2,6 +2,7 @@
 // LICENSE: https://github.com/ruby/openssl/blob/master/LICENSE.txt
 #include "ossl.h"
 
+extern struct RClass *mX509;
 struct RClass *eX509ReqError;
 struct RClass *cX509Req;
 #define SetX509Req(obj, req)                                                                       \

